examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -28,144 +25,378 @@
+
+    SSD can either be used as a dimentionality reduction method or a denoised
+    ‘denoised’ low rank factorization method [2].
 
     Parameters
     ----------
     filt_params_signal : dict
         Filtering for the frequencies of interst.
     filt_params_noise  : dict
         Filtering for the frequencies of non-interest.
-    estimator : str
-        Which covariance estimator to use. Defaults to "oas".
-    n_components  : int, None
-        How many components to keep. Default to 5.
-    subtract_signal : bool
-        Whether to subtract the noise from the signal to enhcane
-        sptial filter.
-    sort_by_spectral_ratio : bool
-        Whether to sort by the spectral ratio. Defaults to True.
-
+    sampling_freq : float
+        sampling frequency (in Hz) of the recordings.
+    estimator : float | str | None (default 'oas')
+        Which covariance estimator to use
+        If not None (same as 'empirical'), allow regularization for
+        covariance estimation. If float, shrinkage is used
+        (0 <= shrinkage <= 1). For str options, estimator will be passed to
+        method to mne.compute_covariance().
+    n_components : int| None (default None)
+        The number of components to decompose the signals.
+        If n_components is None no dimentionality reduction is made, and the
+        transformed data is projected in the whole source space.
+   picks: array| int | None  (default None)
+       Indeces of good-channels. Can be the output of mne.pick_types
+
+   sort_by_spectral_ratio: bool (default True)
+       if set to True, the components are sorted according
+       to the spectral ratio
+       See [1] Nikulin 2011, Eq. (24)
+
+   return_filtered : bool (default False)
+        If return_filtered is True, data is bandpassed and projected onto
+        the SSD components.
+
+   n_fft: int (default None)
+       if sort_by_spectral_ratio is set to True, then the sources will be
+       sorted accordinly to their spectral ratio which is calculated based on
+       "psd_array_welch" function. the n_fft set the length of FFT used.
+       See mne.time_frequency.psd_array_welch for more information
+
+   cov_method_params : TYPE, optional
+        As in mne.decoding.SPoC
+        The default is None.
+   rank : None | dict | ‘info’ | ‘full’
+        As in mne.decoding.SPoC
+        This controls the rank computation that can be read from the
+        measurement info or estimated from the data.
+        See Notes of mne.compute_rank() for details.The default is None.
+        The default is None.
+
+    REFERENCES:
+    [1] Nikulin, V. V., Nolte, G., & Curio, G. (2011). A novel method for
+    reliable and fast extraction of neuronal EEG/MEG oscillations on the basis
+    of spatio-spectral decomposition. NeuroImage, 55(4), 1528-1535.
+    [2] Haufe, S., Dähne, S., & Nikulin, V. V. (2014). Dimensionality reduction
+    for the analysis of brain oscillations. NeuroImage, 101, 583-597.
```

Comment:
Have a look at other functions which print references.
These days we use :footcite: to extract the reference from a bib file.
See #8268 for example.

Reply:
Check it out, you should have access to `doc/references.bib`